Lib/threading.py

```diff
@@ -924,6 +927,7 @@ class is implemented.
         if _HAVE_THREAD_NATIVE_ID:
             self._native_id = None
         self._tstate_lock = None
+        self._join_lock = None
```

Comment:
What about `self._non_joined_finalizer`?

```suggestion
        self._join_lock = None
        self._non_joined_finalizer = None
```